server/eclipse-project/src/main/java/us/freeandfair/corla/model/CountyContestComparisonAudit.java

```diff
@@ -559,70 +560,64 @@ public Integer computeDiscrepancy(final CastVoteRecord the_cvr,
-    // for elections with multiple winners, but will err on the side of being
-    // pessimistic
+    int result = Integer.MAX_VALUE;
     
     // check for overvotes
     final Set<String> acvr_choices = new HashSet<>();
     if (the_acvr_info.choices().size() <= my_contest_result.votesAllowed()) {
       acvr_choices.addAll(the_acvr_info.choices());
     } // else overvote so don't count the votes
     
-    // find the candidates who gained and lost votes in the ACVR
-
-    final Set<String> gained_votes = new HashSet<>(acvr_choices);
-    gained_votes.removeAll(the_cvr_info.choices());
-    final Set<String> winner_gains = new HashSet<>(gained_votes);
-    winner_gains.removeAll(my_contest_result.losers());
-    final Set<String> loser_gains = new HashSet<>(gained_votes);
-    loser_gains.removeAll(my_contest_result.winners());
-
-    final Set<String> lost_votes = new HashSet<>(the_cvr_info.choices());
-    lost_votes.removeAll(acvr_choices);
-    final Set<String> winner_losses = new HashSet<>(lost_votes);
-    winner_losses.removeAll(my_contest_result.losers());
-    final Set<String> loser_losses = new HashSet<>(lost_votes);
-    loser_losses.removeAll(my_contest_result.winners());
-
-    // now, we have several cases:
-    
-    if (gained_votes.isEmpty() && lost_votes.isEmpty()) {
-      // the CVR and ACVR have identical choices and we
-      // can skip the rest of this
-      result = 0;
-    } else if (!winner_gains.isEmpty() && loser_losses.isEmpty()) {
-      // if only winners gained votes and no losers lost votes, 
-      // it's a 1-vote understatement       
-      result = -1;
-    } else if (!winner_gains.isEmpty() && !loser_losses.isEmpty()) {        
-      // if only winners gained votes and any losers lost votes, 
-      // it's a 2-vote understatement       
-      result = -2;
-    } else if (!winner_losses.isEmpty() && loser_gains.isEmpty()) {
-      // if any winners lost votes and no losers gained votes, 
-      // it's a 1-vote overstatement
-      result = 1;
-    } else if (!winner_losses.isEmpty() && !loser_gains.isEmpty()) {
-      // if any winners lost votes and any losers gained votes,
-      // it's a 2-vote overstatement
-      result = 2;
-    } else if (loser_gains.isEmpty()) {
-      // no votes changed for winners, and no losers gained votes,
-      // so losers must have lost votes; if _all_ the losers lost
-      // votes, it'd be a 1-vote understatement
-      if (loser_losses.equals(my_contest_result.losers())) {
-        result = -1;
+    // now, find the minimum pairwise update delta
+    for (final String winner : my_contest_result.winners()) {
+      final int winner_change;
+      if (!the_cvr_info.choices().contains(winner) && acvr_choices.contains(winner)) {
+        // the winner gained a vote
+        winner_change = 1;
+      } else if (the_cvr_info.choices().contains(winner) && !acvr_choices.contains(winner)) {
+        // the winner lost a vote
+        winner_change = -1;
```

Comment:
If a winner who was considered earlier in this loop gained a vote or had a winner_change of 0, this would overwrite that result, perhaps resulting in the wrong outcome when compared with https://archive.org/search.php?query=https%3A%2F%2Fwww.stat.berkeley.edu%2F~stark%2FVote%2FauditTools.htm

Reply:
This would not overwrite that result, because it is factored into the minimization at the end of the second loop. Neither would the one below. However, if it is wrong for some other reason, that would be useful to know.

Reply:
Ah. Yes, I think this is right; because for overstatements it has to widen _every_ pairwise margin, and for understatements it only has to narrow _any_ pairwise margin. I will fix that.